tasktiger/lua/move_task.lua

```diff
@@ -0,0 +1,85 @@
+local function zadd_w_mode(key, score, member, mode)
+    if mode == "" then
+        redis.call('zadd', key, score, member)
+    elseif mode == "nx" then
+        zadd_noupdate({ key }, { score, member })
+    elseif mode == "xx" then
+        zadd_update_existing({ key }, { score, member })
+    elseif mode == "min" then
+        zadd_update_min({ key }, { score, member })
+    elseif mode == "max" then
+        zadd_update_max({ key }, { score, member })
```

Comment:
Though `lt` and `gt` are not supported by Redis 4.0, so we should think whether we can drop support here.

(`nx` and `xx` are supported though)

Reply:
replying just once, applies to all :smile:

two reasons I decided against it:
- NX and XX were added in v3.0.2, LT & GT were added in 6.2.0. we're currently running TT on 3.2 so we can't use LT&GT
- I wanted to keep the implementation as close as possible to the original, just rewritten to Lua. That's because this change is risky as is, I don't want to do optimizations and refactors in the same step as well

Reply:
an example for the 2nd point - the native implementation of NX, XX, GT _could_ be subtly different from what we do in our scripts. It could slip through and work most of the time but cause bugs or corruption down the line.

Reply:
I checked, and `lt`/`gt` were introduced somewhere after Redis 6.0 (probably in Redis 7.0), so we'd need to drop support for Redis 4, 5 and 6. Probably not the best idea.